smac/optimizer/local_bo/boing_subspace.py

```diff
@@ -0,0 +1,167 @@
+        acq_optimizer_local_kwargs: typing.Optional[dict] = None,
+    ):
+        super(BOinGSubspace, self).__init__(
+            config_space=config_space,
+            bounds=bounds,
+            hps_types=hps_types,
+            bounds_ss_cont=bounds_ss_cont,
+            bounds_ss_cat=bounds_ss_cat,
+            model_local=model_local,
+            model_local_kwargs=model_local_kwargs,
+            acq_func_local=acq_func_local,
+            acq_func_local_kwargs=acq_func_local_kwargs,
+            rng=rng,
+            initial_data=initial_data,
+            activate_dims=activate_dims,
+            incumbent_array=incumbent_array,
+        )
+        if bounds_ss_cont is None and bounds_ss_cat is None:
+            self.config_origin = None  # type: ignore
+        else:
+            self.config_origin = "BOinG"
+        if isinstance(self.model, GloballyAugmentedLocalGP):
+            num_inducing_points = min(max(min(2 * len(self.activate_dims_cont), 10), self.model_x.shape[0] // 20), 50)
+            self.model.update_attribute(num_inducing_points=num_inducing_points)
+
+        subspace_acq_func_opt_kwargs = {
+            "acquisition_function": self.acquisition_function,
+            "config_space": self.cs_local,  # type: ignore[attr-defined] # noqa F821
+            "rng": np.random.RandomState(self.rng.randint(1, 2**20)),
+        }
+
+        if isinstance(acq_optimizer_local, AcquisitionFunctionMaximizer):
+            # we copy the attribute of the local acquisition function optimizer but replace it with our local model
+            # setting. This helps a better exploration in the beginning.
+            for key in inspect.signature(acq_optimizer_local.__init__).parameters.keys():  # type: ignore[misc]
+                if key == "self":
+                    continue
+                elif key in subspace_acq_func_opt_kwargs:
+                    continue
+                elif hasattr(acq_func_local, key):
+                    subspace_acq_func_opt_kwargs[key] = getattr(acq_func_local, key)
+            self.acq_optimizer_local = type(acq_optimizer_local)(**subspace_acq_func_opt_kwargs)
+        else:
+            if acq_optimizer_local is None:
+                acq_optimizer_local = LocalAndSortedRandomSearch  # type: ignore
+                # Here are the setting used by squirrel-optimizer
+                # https://github.com/automl/Squirrel-Optimizer-BBO-NeurIPS20-automlorg/blob/main/squirrel-optimizer/smac_optim.py
+                n_sls_iterations = {
+                    1: 10,
+                    2: 10,
+                    3: 10,
+                    4: 10,
+                    5: 10,
+                    6: 10,
+                    7: 8,
+                    8: 6,
+                }.get(len(self.cs_local.get_hyperparameters()), 5)
+                subspace_acq_func_opt_kwargs.update({"n_steps_plateau_walk": 5, "n_sls_iterations": n_sls_iterations})
```

Comment:
since this might be changed later, consider putting (and documenting) this into the scenario/smac options.

Reply:
As this is only a sub-argument from BOinG and not used anywhere else. I would prefer to keep it here. Additionally, n_sls_iterations is determined by the number of HPs in the local search space, which can be changed dynamically as optimization continues